000-vip-init.php

```diff
@@ -288,5 +288,29 @@
 	add_filter( 'wp_sitemaps_enabled', '__return_false' );
 }
 
+// We use HyperDB, so DB_* constants are not automatically defined
+if ( ! defined( 'DB_NAME' ) && ! defined( 'DB_HOST' ) && ! defined( 'DB_PASSWORD' ) && ! defined( 'DB_USER' ) ) {
+	global $db_servers;
```

Comment:
Can we move this logic to a function and add some tests for it?

We could also simplify it by fetching the db details from `$wpdb->hyper_servers` (https://github.com/Automattic/HyperDB/blob/trunk/db.php#L266-L268), although, it would be nicer/safer if hyperdb had a `get` method for that.

Reply:
Moved to a function now...couldn't quite figure out how to mock hyperdb with PHPUnit atm since it uses the native $wpdb.

Reply:
What do you think if:

```diff
-function vip_define_db_constants() {
+function vip_define_db_constants( $db ) {
...
-add_action( 'vip_loaded', 'vip_define_db_constants' );
+add_action( 'vip_loaded', function () {
+    global $wpdb;
+    vip_define_db_constants( $wpdb );
+} );
```

?

Then we can create a helper class with `get_hyper_servers()` and test the function.

Reply:
But we won't be unable to mock `defined()` in `if ( defined( 'DB_NAME' ) || defined( 'DB_HOST' ) || defined( 'DB_PASSWORD' ) || defined( 'DB_USER' ) ) {` because the function is not in a namespace. We will probably have to move it.

Reply:
If you are OK with that, I can implement this.